agent.go

```diff
@@ -389,10 +409,13 @@ func NewAgent(config *AgentConfig) (*Agent, error) {
 		a.taskLoopInterval = config.taskLoopInterval
 	}
 
-	if config.CandidateTypes == nil || len(config.CandidateTypes) == 0 {
```

Comment:
We still want to use the user provided `CandidateTypes` if they passed in. I just want to do an assert when `lite` is chosen. I think the user should be explicit when using `lite` so they fully understand what is happening. We can loosen this restriction later, but I just don't want inputs to conflict.

```
if a.lite && (len(a.candidateTypes) != 1 || a.candidateTypes[0] != CandidateTypeHost) {
  return nil ErrLiteUsingNonHostCandidates
}
```